iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -955,9 +956,9 @@ public override async Task SendTwinPatchAsync(TwinCollection reportedProperties,
                                 Debug.WriteLine("ConnectException trying to connect to {0}: {1}", address.ToString(), ex.ToString());
                                 return true;
                             }
+
```

Comment:
the return in catch logic causes another bug. If there is error throws in the first call to ConnectAsync, there is no channel returned. We found this when trying the new SDK 1.17.1.  So it seems we need keep trying connection when exception happens

Reply:
@abhipsaMisra PTAL.

@adashen Please add more details: if the lambda passed to Handle returns `true` it means that the exception was handled and the `foreach` statement should continue (retry). If the return is false then we throw that exception.
https://msdn.microsoft.com/en-us/library/system.aggregateexception.handle(v=vs.110).aspx#Anchor_2